app/controllers/events_controller.rb

```diff
@@ -0,0 +1,29 @@
+class EventsController < ApplicationController
+  before_action :authenticate_user!, except: %i[index]
+
+  def index
+    @events = Event.all
+    @future_events = @events.upcoming
+    @past_events = @events.past
```

Comment:
- [ ] Refactor this code: Currently, the implementation seems difficult to understand. I believe it should be similar to this:
@future_events = Event.upcoming
@past_events = Event.past